scripts/release/asset_publish.py

```diff
@@ -277,33 +286,66 @@ def _str2bool(v: str) -> bool:
             )
 
             # Handle specific asset types
-            if asset.type in [assets.AssetType.COMPONENT, assets.AssetType.ENVIRONMENT]:
-                # Assemble command
-                cmd = assemble_command(
-                    asset.type.value, str(asset.spec_with_path),
-                    registry_name, final_version, resource_group, workspace, debug_mode)
-                # Run command
-                run_command(cmd, failure_list, debug_mode)
+            if asset.type == assets.AssetType.COMPONENT:
+                # load component and check if environment exists
+                logger.print(f"spec's path: {asset.spec_with_path}")
+                component = load_component(asset.spec_with_path)
+                env = component.environment
+                ws_env_pattern = re.compile("^(.+):(.+)$")
```

Comment:
Shouldn't this be:
```suggestion
                ws_env_pattern = re.compile("^azureml:(.+):(.+)$")
```

Reply:
`azureml:` prefix is omitted when I call load_component. So for a component with `environment: azureml:AzureML-sklearn-1.0-ubuntu20.04-py38-cpu@latest` it captures env as below: 

![image](https://user-images.githubusercontent.com/61145377/218708375-4488b522-5c42-4f94-801c-78249004fcb3.png)